test/e2e/network_policy.go

```diff
@@ -0,0 +1,440 @@
+		}
+		if resp.StatusCode != http.StatusOK {
+			framework.Failf("Unexpected error code, expected 200, got, %v (%v)", resp.StatusCode, resp)
+		}
+	})
+
+	It("should isolate containers when NetworkIsolation is enabled [Policy]", func() {
+		runIsolatedToBidirectionalTest(f)
+	})
+})
+
+// Run a test that starts with fully isolated namespaces and adds policy objects to
+// allow bi-directional (and mono-directional) traffic between namespaces.
+func runIsolatedToBidirectionalTest(f *framework.Framework) {
+	// These tests use two namespaces.  A single namespace is created by
+	// default.  Create another and store both separately for clarity.
+	nsA := f.Namespace
+	nsB, err := f.CreateNamespace(f.BaseName+"-b", map[string]string{
+		"e2e-framework": f.BaseName + "-b",
+	})
+	Expect(err).NotTo(HaveOccurred())
+
+	// Turn on isolation
+	setNetworkIsolationAnnotations(f, nsA, true)
+	setNetworkIsolationAnnotations(f, nsB, true)
+
+	// Get the available nodes.
+	nodes, err := framework.GetReadyNodes(f)
+	framework.ExpectNoError(err)
+
+	if len(nodes.Items) == 1 {
+		// in general, the test requires two nodes. But for local development, often a one node cluster
+		// is created, for simplicity and speed. (see issue #10012). We permit one-node test
+		// only in some cases
+		if !framework.ProviderIs("local") {
+			framework.Failf(fmt.Sprintf("The test requires two Ready nodes on %s, but found just one.", framework.TestContext.Provider))
+		}
+		framework.Logf("Only one ready node is detected. The test has limited scope in such setting. " +
+			"Rerun it with at least two nodes to get complete coverage.")
+	}
+
+	// Create service A and B.  These are really just used
+	// for pod discovery by the net monitor containers.
+	serviceA := createService(f, nsA, serviceAName)
+	serviceB := createService(f, nsB, serviceBName)
+
+	// Clean up services
+	defer func() {
+		By("Cleaning up the service A")
+		if err = f.Client.Services(nsA.Name).Delete(serviceA.Name); err != nil {
+			framework.Failf("unable to delete svc %v: %v", serviceA.Name, err)
+		}
+	}()
+	defer func() {
+		By("Cleaning up the service B")
+		if err = f.Client.Services(nsB.Name).Delete(serviceB.Name); err != nil {
+			framework.Failf("unable to delete svc %v: %v", serviceB.Name, err)
+		}
+	}()
+
+	By("Creating a webserver (pending) pod on each node")
+	podAName, podBNames := launchNetMonitorPods(f, nsA, nsB, nodes)
+
+	// Deferred clean up of the pods.
+	defer func() {
+		By("Cleaning up the webserver pods")
+		if err = f.Client.Pods(nsA.Name).Delete(podAName, nil); err != nil {
+			framework.Logf("Failed to delete pod %s: %v", podAName, err)
+		}
+		for _, podName := range podBNames {
+			if err = f.Client.Pods(nsB.Name).Delete(podName, nil); err != nil {
+				framework.Logf("Failed to delete pod %s: %v", podName, err)
+			}
+		}
+	}()
+
+	// Wait for all pods to be running.
+	By(fmt.Sprintf("Waiting for pod %q to be running", podAName))
```

Comment:
Is %q the right formatter here?  

`%q	a single-quoted character literal safely escaped with Go syntax.`